CMSgov/mint-app
# [EASI-2511] Operational Needs Subtasks -- adding, removing, and managing

# EASI-2511

## Changes and Description

- Add new `subtask` i18n file
- Add `CreationOperationalSolutionSubtasks`, `DeleteOperationalSolutionSubtask`, `UpdateOperationalSolutionSubtasks` queries and types
- Updated `SolutionDetails` to use real Subtask data
- Created new `add-subtasks` and `manage-subtask` pages



## How to test this change

1. View details of an Operational need with a solution
2. See an empty Subtask table 
![image](https://user-images.githubusercontent.com/2349702/225395065-5e29c3b8-b182-4794-9f00-7e60f78ff810.png)
    i. `No subtasks added yet` is in the `To do` column
    
    ii. Only `Add subtasks` button available
    
    iii. `Manage subtasks` button is mia
3. Click add a new subtask button -- should push you to a new page
4. Basic Add subtask page to match Figma
5. Clicking `Add another subtask` should render a new input box and checkboxes underneath
  i. I talked with Natasha and she said she is good with this implementation (rather than the initial push to another page design)
6. `Add subtask` button is disabled until all subtask names are filled in
7. Fill in ames and hit `Add subtask`
8. Ensure that the new subtasks are displaying on table
9. Manage subtask button is now available
10. Play with the `manage a subtask` page
11. try to break things
  


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
